dispatch.go

```diff
@@ -40,13 +40,20 @@ func (d *dispatch) Call(ctx context.Context, name string, arg interface{}, res i
 	profiler := d.log.StartProfiler("call %s", name)
 	defer profiler.Stop()
 
+	select {
+	case <-d.stopCh:
```

Comment:
Yeah but I'd rather not even attempt to encode if we're supposed to be stopped. The key here is that once `Close` is called, we won't try to dispatch at all.